### PR TITLE
Import code

### PR DESCRIPTION
**Pull request by doringutsu**

### CODE REVIEW

File: app/models/student.rb [hunk 1]
```diff
@@ -2,12 +2,31 @@
...
+#  first_name          :string           not null
+#  last_name           :string           not null
+#  student_group_id    :integer          not null
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#  nationality         :string
+#  came_from           :string
+#  came_at             :date
+#  came_year           :integer
+#  idnp                :string
```
Comment (hpetru) [inline app/models/student.rb:12]: idnp_token

File: app/models/student.rb [hunk 2]
```diff
@@ -2,12 +2,31 @@
...
+#  address_id          :string
```
Comment (hpetru) [inline app/models/student.rb]: trebuie sa fie integer

File: app/models/student.rb [hunk 3]
```diff
@@ -2,12 +2,31 @@
...
+#  address_house       :string
+#  address_appartment  :string
+#  father_last_name    :string
+#  father_first_name   :string
+#  father_phone_number :string
+#  mother_last_name    :string
+#  mother_first_name   :string
+#  mother_phone_number :string
+#  mother_email        :string
+#  gender              :string
```
Comment (hpetru) [inline app/models/student.rb]: trebuie de adaugat not null

File: app/models/student.rb [hunk 4]
```diff
@@ -2,12 +2,31 @@
...
+#  inside_code         :string
```
Comment (hpetru) [inline app/models/student.rb]: inside_code_token

File: app/models/student.rb [hunk 5]
```diff
@@ -2,12 +2,31 @@
...
+#  birthday            :date
```
Comment (hpetru) [inline app/models/student.rb:25]: Not null

File: app/models/student.rb [hunk 6]
```diff
@@ -2,12 +2,31 @@
...
-#  student_group_id :integer          not null
-#  created_at       :datetime         not null
-#  updated_at       :datetime         not null
+#  id                  :integer          not null, primary key
+#  first_name          :string           not null
+#  last_name           :string           not null
+#  student_group_id    :integer          not null
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#  nationality         :string
```
Comment (hpetru) [inline app/models/student.rb]: Not null

File: app/models/student.rb [hunk 7]
```diff
@@ -2,12 +2,31 @@
...
+#  came_from           :string
+#  came_at             :date
+#  came_year           :integer
```
Comment (hpetru) [inline app/models/student.rb]: Salvam datele in `came_at`

File: db/migrate/20160429055543_create_join_table.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+class CreateJoinTable < ActiveRecord::Migration
+  def change
+    create_join_table :subjects, :teachers do |t|
+      # t.index [:subject_id, :teacher_id]
```
Comment (hpetru) [inline db/migrate/20160429055543_create_join_table.rb:4]: De ce-i comentat ?

File: db/migrate/20160429053951_create_colleges.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+class CreateColleges < ActiveRecord::Migration
+  def change
+    create_table :colleges do |t|
```
Comment (hpetru) [inline db/migrate/20160429053951_create_colleges.rb:3]: Nu am vazut model pentru tabelul ista

Comment (doringutsu) [inline db/migrate/20160429053951_create_colleges.rb:3]: A fost redenumit in `institutions`

File: app/helpers/application_helper.rb [hunk 10]
```diff
@@ -1,2 +1,17 @@
 module ApplicationHelper
+  def full_name(obj)
```
Comment (hpetru) [inline app/helpers/application_helper.rb]: sterge, treb fiecare model sa defineasca acest atribut

File: app/helpers/application_helper.rb [hunk 11]
```diff
@@ -1,2 +1,17 @@
...
+    obj.last_name + ' ' + obj.first_name
+  end
+
+  def study_year_from(date)
```
Comment (hpetru) [inline app/helpers/application_helper.rb]: treb de sters ...

File: app/helpers/application_helper.rb [hunk 12]
```diff
@@ -1,2 +1,17 @@
...
+    date.month > 8 ? date.year : date.year - 1
+  end
+
+  def current_study_year
```
Comment (hpetru) [inline app/helpers/application_helper.rb]: -

File: app/helpers/application_helper.rb [hunk 13]
```diff
@@ -1,2 +1,17 @@
...
+    study_year_from Date.today
+  end
+
+  def current_year_from(grade)
```
Comment (hpetru) [inline app/helpers/application_helper.rb]: -

File: app/models/address.rb [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+#
+#  id         :integer          not null, primary key
+#  street     :string           not null
+#  city_id    :integer          not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+class Address < ActiveRecord::Base
+  validates_presence_of :city
```
Comment (hpetru) [inline app/models/address.rb]:
```
validates(
  :city,
  :street,
  present: true
)
```

File: db/schema.rb [hunk 15]
```diff
@@ -75,11 +92,30 @@
...
-    t.string   "last_name",        null: false
-    t.integer  "student_group_id", null: false
-    t.datetime "created_at",       null: false
-    t.datetime "updated_at",       null: false
+    t.string   "first_name",          null: false
+    t.string   "last_name",           null: false
+    t.integer  "student_group_id",    null: false
+    t.datetime "created_at",          null: false
+    t.datetime "updated_at",          null: false
+    t.string   "nationality"
```
Comment (hpetru) [inline db/schema.rb]: Not null

File: db/schema.rb [hunk 16]
```diff
@@ -75,11 +92,30 @@
...
+    t.string   "came_from"
+    t.date     "came_at"
+    t.string   "idnp"
```
Comment (hpetru) [inline db/schema.rb:107]: rename to `idnp_token`

File: db/schema.rb [hunk 17]
```diff
@@ -75,11 +92,30 @@
...
+    t.string   "address_house"
+    t.string   "address_appartment"
+    t.string   "father_last_name"
+    t.string   "father_first_name"
+    t.string   "father_phone_number"
+    t.string   "mother_last_name"
+    t.string   "mother_first_name"
+    t.string   "mother_phone_number"
+    t.string   "mother_email"
+    t.string   "gender"
```
Comment (hpetru) [inline db/schema.rb]: not null

File: db/schema.rb [hunk 18]
```diff
@@ -75,11 +92,30 @@
...
+    t.string   "inside_code_token"
```
Comment (hpetru) [inline db/schema.rb]: not null

File: db/schema.rb [hunk 19]
```diff
@@ -31,6 +31,23 @@
...
+  end
+
+  create_table "cities", force: :cascade do |t|
+    t.string   "name",       null: false
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  create_table "institutions", force: :cascade do |t|
+    t.string "name"
```
Comment (hpetru) [inline db/schema.rb]: not null

File: app/controllers/student_groups_controller.rb [hunk 20]
```diff
@@ -20,6 +20,10 @@ def edit
     @student_group = StudentGroup.find(params[:id])
   end
 
+  def show
```
Comment (hpetru) [inline app/controllers/student_groups_controller.rb]: remove `show` method and related views

File: app/models/student.rb [hunk 21]
```diff
@@ -19,18 +38,21 @@ class Student < ActiveRecord::Base
...
     presence: true
   )
 
   belongs_to :student_group
+  belongs_to :address
   has_many :semester_grades, class_name: StudentSemesterGrade
   has_many :anual_grades, class_name: StudentAnualGrade
   has_many :thesis_grades, class_name: StudentThesisGrade
-
+  belongs_to :came_from, class_name: Institution.to_s
```
Comment (hpetru) [inline app/models/student.rb]: without `.to_s`

File: app/models/student_group.rb [hunk 22]
```diff
@@ -20,12 +20,11 @@ class StudentGroup < ActiveRecord::Base
   validates(
     :promotion,
     :suffix,
-    :profile_slug,
     :main_teacher,
     presence: true
   )
   validates :suffix, length: { maximum: 1 }
-
+  has_many :students
```
Comment (hpetru) [inline app/models/student_group.rb:28]: Please add a blank line before

File: app/models/teacher.rb [hunk 23]
```diff
@@ -8,31 +8,39 @@
...
-    'third_degree' => 'third_degree',
-    'superior_degree' => 'superior_degree'
+    doctorate: 'doctorate',
+    first_degree: 'first_degree',
+    second_degree: 'second_degree',
+    third_degree: 'third_degree',
+    superior_degree: 'superior_degree'
+  }
+  enum gender: {
+    m: 'm',
```
Comment (hpetru) [inline app/models/teacher.rb]: rename to `male: 'male'`

File: db/migrate/20160510071417_set_default_nationality.rb [hunk 24]
```diff
@@ -0,0 +1,5 @@
+class SetDefaultNationality < ActiveRecord::Migration
+  def change
+    change_column :students, :nationality, :string, default: 'mold', null: false
```
Comment (hpetru) [inline db/migrate/20160510071417_set_default_nationality.rb]: should be updated

**Pull request merged**